cfg/libcerror.cfg

```diff
@@ -0,0 +1,128 @@
+<?xml version="1.0"?>
+<!-- Official repository: https://github.com/libyal/libcerror -->
+<!-- Typically included by: #include <libcerror.h> -->
+<def format="2">
+  <!-- int libcerror_error_initialize( libcerror_error_t **error,
+                                       int error_domain,
+                                       int error_code ); -->
+  <function name="libcerror_error_initialize">
+    <noreturn>false</noreturn>
+    <returnValue type="int"/>
+    <arg nr="1">
+      <not-null/>
+      <not-uninit/>
+    </arg>
+    <arg nr="2">
+      <not-uninit/>
+    </arg>
+    <arg nr="3">
+      <not-uninit/>
+    </arg>
+  </function>
+  <!-- void libcerror_error_set( libcerror_error_t **error,
+                                 int error_domain,
+                                 int error_code,
+                                 const char *format_string,
+                                 ... ); -->
+  <function name="libcerror_error_set">
+    <noreturn>false</noreturn>
+    <returnValue type="void"/>
+    <arg nr="1">
+      <not-null/>
+      <not-uninit/>
+    </arg>
+    <arg nr="2">
+      <not-uninit/>
+    </arg>
+    <arg nr="3">
+      <not-uninit/>
+    </arg>
+    <formatstr/>
+    <arg nr="4">
+      <not-null/>
+      <not-uninit/>
+      <formatstr/>
+      <strz/>
+    </arg>
+  </function>
+  <!-- void libcerror_error_free( libcerror_error_t **error ); -->
+  <function name="libcerror_error_free">
+    <noreturn>false</noreturn>
+    <returnValue type="void"/>
+    <arg nr="1">
+      <not-uninit/>
+    </arg>
+  </function>
+  <!-- int libcerror_error_resize( libcerror_internal_error_t *internal_error ); -->
+  <function name="libcerror_error_resize">
+    <noreturn>false</noreturn>
+    <returnValue type="int"/>
+    <arg nr="1">
+      <not-null/>
+      <not-uninit/>
+    </arg>
+  </function>
+  <!-- int libcerror_error_fprint( libcerror_error_t *error, FILE *stream ); -->
+  <!-- int libcerror_error_backtrace_fprint( libcerror_error_t *error, FILE *stream ); -->
+  <function name="libcerror_error_fprint,libcerror_error_backtrace_fprint">
+    <noreturn>false</noreturn>
+    <returnValue type="int"/>
```

Comment:
Can the specification “`<use-retval/>`” become relevant for this function description?

Reply:
No, would produce false positives i think.

Reply:
Will interesting warnings be occasionally generated?

Reply:
> Will interesting warnings be occasionally generated?

yes it would.

please stop recommending this, you will never follow the cppcheck philosophy.

Reply:
Do you know circumstances under which you would care more also for file output failures?

Reply:
I guess we would need additional heuristics to avoid false positives.

Reply:
If there are 90 real bugs...  we rather report 10 warnings without false positives; than 100 warnings with 10 false positives.

I think it might be possible to invent some heuristics to catch some of those 90 bugs without reporting any false positives, but we will never be near 90.

I envision that it would be possible to add some attributes to the `<use-retval/>` that specify the extra heuristics. Imagine that the return value _must_ be taken if argument 3 is NULL. Then a pretty generic way to specify that would be:

    <use-retval>
        <if-equal arg="3" value="0"/>
    </use-retval>

Reply:
I imagine that there is a need for further adjustments of the data model for programming interface descriptions.
But progress depends on your mood to discuss corresponding changes.

Reply:
I am not in the mood to discuss it. Feel free to suggest something concrete in a forum or something. Something like my suggestion where you show what an actual xml configuration could look like. I think it would help if you also mention some actual function that would use such configuration.

Reply:
I am curious if your development attention can become better also for these software areas.

Reply:
I have the feeling you use some kind of AI tool to continue discussions for ever and obfuscate replies. Probably developed by you.

Reply:
Your impression is inappropriate here.